### PR TITLE
Update error handling in Admin Server

### PR DESCRIPTION
**Pull request by jordanschalm**
⚠️  This PR extends and depends on https://github.com/onflow/flow-go/pull/3369

This PR defines expected error types for admin command implementations, and updates existing Validator functions to make use of the validation sentinel error. It also updates the admin server to treat unexpected errors as an internal failure and log any unexpected errors.

### CODE REVIEW

**Review by AlexHentschel — APPROVED** [review]
very nice. Thanks you so much for substantially improving the error handing of the admin commands and adding lots of documentation. 🎖️

File: admin/commands/storage/read_transactions.go [hunk 1]
```diff
@@ -60,23 +60,26 @@ func usageErr(msg string) error {
...
 func findUint64(input map[string]interface{}, field string) (uint64, error) {
 	data, ok := input[field]
 	if !ok {
-		return 0, usageErr(fmt.Sprintf("%s not set", field))
+		return 0, admin.NewInvalidAdminReqErrorf("missing required field '%s'", field)
 	}
 	val, err := parseN(data)
 	if err != nil {
-		return 0, usageErr(fmt.Sprintf("%s must be a uint64 value, but got %v: %v", field, data, err))
+		return 0, admin.NewInvalidAdminReqErrorf("invalid 'n' field: %w", err)
```
Comment (zhangchiqing) [inline admin/commands/storage/read_transactions.go:71]: better to print the field as well

**Review by zhangchiqing — APPROVED** [review]

File: admin/commands/storage/read_transactions.go (hunk 1 above)
Comment (jordanschalm) [reply admin/commands/storage/read_transactions.go:66]: Done in https://github.com/onflow/flow-go/pull/3391/commits/10bc2b1e60723fdbf79d926c382d9d205f4b032f

**Comment by jordanschalm** [comment]
bors merge

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**